### DISCUSSION TITLE
What about mongodb as default package?

### DISCUSSION THREAD

**Discussion by rhadamez** [adonisjs/core#1748] [Ideas]
Hey!

In previous versions of adonis, we have not official lucid mongodb, it is not official, but works. **Now, we are in 2020 with Adonis 5**, I've been thinking that Adonis could add a nosql database to its packages as default.

What do you think guys? It would be great to the framework.

**Comment by avidcoder123** [comment]
I agree that NoSQL would be nice with Adonis, but @thetutlage has made clear that since he is not familiar with NoSQL, he cannot create a NoSQL ORM/ODM or extend Lucid to be compatible with it.

**Comment by RomainLanz** [comment]
Hey all! 👋 

This question has been answered dozen of times.

Quick answer to *"When Lucid will support MongoDB"*: **Never**

> Now, we are in 2020 with Adonis 5

The year we are in will not change that NoSQL **shouldn't be used** as a primary database.

Now, if you want to use MongoDB in your application, you are free to install any packages like `mongoose`.
**AdonisJS doesn't restrict you from installing a driver (or any packages).**

Also, I want to make something really clear here. You are saying you want Lucid for MongoDB. Lucid is an **ORM**, meaning it helps you to work with relational data. If you are storing relational data inside a NoSQL database, you are missing the point of this technology and you have made a poor choice for your database system.

**Comment by Zizaco** [comment]
I agree that including MongoDB as an option in Lucid would **not** be ideal, given that Lucid is SQL-based.

Nevertheless, the  MongoDB provider for Laravel is now part of their official docs:
https://laravel.com/docs/11.x/mongodb
https://github.com/laravel/docs/blob/11.x/mongodb.md _(Oct 31, 2024)_

@RomainLanz @thetutlage would you be open to having an "official"/recommended MongoDB provider for Adonis (to be listed on the website/docs like Redis)?

> NoSQL shouldn't be used as a primary database

I hope you don't mind me mentioning this, but MongoDB has evolved considerably since the 2010s. Many important workloads use it as a primary database. [1](https://stripe.com/blog/how-stripes-document-databases-supported-99.999-uptime-with-zero-downtime-data-migrations), [2](https://medium.com/@m.priyanka.raja/ebays-transformation-with-mongodb-52d0cd9245f), [3](https://www.mongodb.com/blog/post/transforming-payments-volante-mongodb-modern-cloud-solution#:~:text=66%25%20of%20U.S.%20commercial%20deposits), [4](https://www.mongodb.com/solutions/customer-case-studies/wells-fargo), [5](https://commercetools.com/infrastructure-partners/mongodb), [6](https://pages.awscloud.com/apn-tv-826.html?trk=apn-tv-carousel), [7](https://pages.awscloud.com/apn-tv-796.html?trk=apn-tv-carousel), [8](https://www.mongodb.com/solutions/customer-case-studies/shopline)
